public/scripts/ui.js

```diff
@@ -1299,37 +1298,52 @@ class Base64ZipDialog extends Dialog {
         this.$pasteBtn.innerText = "Processing...";
     }
 
-    async processClipboard(type) {
-        if (!navigator.clipboard.readText) {
-            Events.fire('notify-user', 'This feature is not available on your browser.');
-            console.log("navigator.clipboard.readText() is not available on your browser.")
-            this.hide();
-            return;
+    preparePasting(type) {
+        if (navigator.clipboard.readText) {
+            this.$pasteBtn.innerText = `Tap here to paste ${type}`;
+            this.$pasteBtn.addEventListener('click', _ => this.processClipboard(type));
+        } else {
+            console.log("`navigator.clipboard.readText()` is not available on your browser.\nOn Firefox you can set `dom.events.asyncClipboard.readText` to true under `about:config` for convenience.")
+            this.$pasteBtn.setAttribute('hidden', '');
+            this.$fallbackTextarea.setAttribute('placeholder', `Paste here to send ${type}`);
+            this.$fallbackTextarea.removeAttribute('hidden');
+            this.$fallbackTextarea.addEventListener('input', e => this.processInput(e, type));
```

Comment:
Nit: either remove the event listener in `processClipboard`, declare the listener in the constructor, or use `once`.

Reply:
Just curious, is this mostly for tidiness or is there any consequence if I won’t remove it? For now this method is not ever called twice or is it?
Probably best to remove it anyways, especially if I add a possibility to overtake a session: #42

Reply:
Yeah, I figure just good practice to avoid creating a lot of duplicate event listeners. If that ever happens that'd lead to some odd bugs with multiple calls to `processInput`.

Reply:
Did not know about the once option. Nice! I needed to remove the listeners manually though, otherwise a user could not interact anymore when accidentally clicking away of the btn or typing sth / pasting sth in the input that is not base64 encoded. Is added now and will be merged in a sec. Thanks for your help!